modules/clientmanagements/client/controllers/clientinventory.client.controller.js

```diff
@@ -113,63 +180,94 @@
+          if (vm.inventorymanagements[w].upc === $scope.serial[i].upc) {
+            invResult = w;
             break;
           }
         }
-        if (invResult === -1 && clientResult === -1) {
-          alert("That Client and UPC don't exist");
-        } else if (invResult === -1) {
-          alert("That UPC doesn't exist");
-        } else if (clientResult === -1) {
-          alert("That Client doesn't exist");
-        } else {
-          // client and item exist, now check if client has that item
-          var alreadyHas = false;
-          for (i = 0; i < vm.clientmanagements[clientResult].inventory.length; i++) {
-            if (vm.clientmanagements[clientResult].inventory[i].upc === vm.inventorymanagements[invResult].upc) {
-              // client already has this, now decrement by 1 and check if item should be removed
-              vm.clientmanagements[clientResult].inventory[i].qty -= 1;
-              if (vm.clientmanagements[clientResult].inventory[i].qty === 0) {
-                // remove this item from their inventory
-                vm.clientmanagements[clientResult].inventory.splice(i, 1);
-              }
-              alreadyHas = true;
+        // clear upc field
+        $scope.serial[i] = null;
+        $scope.qty[i] = null;
+        for (v = 0; v < clientFields; v++) {
+          var clientInfo = $scope.nameAndEmail[v].split(" --- ");
+          var clientResult = -1;
+          for (w = 0; w < vm.clientmanagements.length; w++) {
+            if (vm.clientmanagements[w].name === clientInfo[0] && vm.clientmanagements[w].email === clientInfo[1]) {
+              clientResult = w;
               break;
             }
           }
-          if (!alreadyHas) {
-            // client doesn't have this item, nothing to transfer
-            alert("Client doesn't have this item");
+          if (invResult === -1 && clientResult === -1) {
+            alert("At least one Client and UPC don't exist");
             return;
+          } else if (invResult === -1) {
+            alert("At least one UPC doesn't exist");
+            return;
+          } else if (clientResult === -1) {
+            alert("At least one Client doesn't exist");
+            return;
+          } else {
+            // client and item exist, now check if client has that item
+            var alreadyHas = false;
+            var notEnough = false;
+            for (w = 0; w < vm.clientmanagements[clientResult].inventory.length; w++) {
+              if (vm.clientmanagements[clientResult].inventory[w].upc === vm.inventorymanagements[invResult].upc) {
+                // client already has this, now decrement by qty and check if item should be removed
+                if (vm.clientmanagements[clientResult].inventory[w].qty < quant) {
+                  // client doesn't have >= $scope.qty[i]
+                  alert(clientInfo[0]+" doesn't have that many, skipping to next client if there is another");
```

Comment:
Will this alert box stop the processing of a list of clients? Do we want to push this to the end of the move function?

Reply:
Unless it has a return after it, this won't stop the processing. I'm pretty sure...

Reply:
Ive Tested it and Im pretty sure it doesn't stop. It will just move on to the next client if there is one. I'll test again to be sure, or if you ever are curious about something in my branch, just pull it down, check it out, and test it for yourself.